### PR TITLE
use correct username for systemd service in nixos

### PR DESCRIPTION
**Pull request by mrVanDalo**

### CODE REVIEW

File: nixos/default.nix [hunk 1]
```diff
@@ -40,7 +40,7 @@ in
         after = [ "nix-daemon.socket" ];
 
         serviceConfig = {
-          User = username;
+          User = config.users.users.${username}.name;
```
Comment (rycee) [inline nixos/default.nix]: Might be slightly more idiomatic to use `usercfg.home.username` here.

**Comment by rycee** [comment]
Good catch! 🙂

I added a small comment about using `home.username` instead. Also, please change the commit message to "nixos: use correct username for systemd service". We typically prefix the commit message with the affected component.

File: nixos/default.nix (hunk 1 above)
Comment (mrVanDalo) [reply nixos/default.nix]: is that important to you? I like it this way a bit better, because this is how `home.username` is defined as well.

**Comment by rycee** [comment]
Thanks! Rebased to master in fbdb5beb5941f79fc38c5c3d92b0029c4aa40ab5.

**Pull request closed without merging**